src/client/mod.rs

```diff
@@ -59,9 +78,36 @@ impl InfluxDbClient {
         InfluxDbClient {
             url: url.to_string(),
             database: database.to_string(),
+            auth: None,
         }
     }
 
+    /// Add authentication/authorization information to [`InfluxDbClient`](crate::client::InfluxDbClient)
+    ///
+    /// # Arguments
+    ///
+    /// * username: The Username for InfluxDB.
+    /// * password: THe Password for the user.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use influxdb::client::InfluxDbClient;
+    ///
+    /// let _client = InfluxDbClient::new("http://localhost:9086", "test").with_auth("admin", "password");
+    /// ```
+    pub fn with_auth<'a, S1, S2>(mut self, username: S1, password: S2) -> Self
```

Comment:
Could we write a test for this? Not sure how / what to assert...

Reply:
I added some tests